src/Keboola/GoogleSheetsWriter/Sheet.php

```diff
@@ -126,13 +124,22 @@ private function appendAction(array $sheet, Table $inputTable): array
         $paginator = new Paginator($inputTable);
         $hasHeader = $this->hasHeader($sheet);
 
-        foreach ($paginator->pages() as $values) {
+        foreach ($paginator->pages() as $offset => $values) {
             // if sheet already contains header, strip header from values to be uploaded
-            if ($paginator->getOffset() === 1 && $hasHeader) {
+            if ($offset === 1 && $hasHeader) {
```

Comment:
Když budeš mít objekt, tak tady máš krásně `$chunk->isFirstChunk()` třeba... Mě by třeba bez kontextu tady zaskočilo, proč to není od nuly? (po tom co jsem to celé četl vím, že je to proto, že to pak používáš v 'A1' např.). Nechci zbytečně ten review protahovat, protože je to dostatečně dobré, ale přijde mi důležité ten feedback na to dát, když si myslím, že by to bylo ve výsledku lepší.

Reply:
To je tiez dobry postreh s tym "isFirst". Jasne, ja s tym nemam problem, prave ze v tomto pripade som diskusii otvoreny. Len ma napadlo ten "offset" spravit ako kluc a zaujimalo ma co na to povies ;) Spravim to ako samostatnu triedu teda. Jo na tom mi vadi vlastne len to, ze ten generator nema ziadny "typ hint" na tu triedu ktoru vrati ten "yield", takze potom ti PHPStorm nenapoveda metody tej "Page"..

Reply:
Vynutit to nejde, ale můžeš mít `@return \Generator|Page[]` a to sežere PhpStan i PhpStorm AFAIK